lib/src/main/kotlin/starknet/data/types/transactions.kt

```diff
@@ -79,12 +82,44 @@ class BlockHashOrTagSerializer() : KSerializer<BlockHashOrTag> {
 
 @Serializable
 data class InvokeFunctionPayload(
-    @SerialName("function_invocation") val invocation: Call,
+    @SerialName("function_invocation")
+    val invocation: Call,
+
     val signature: Signature?,
-    @SerialName("max_fee") val maxFee: Felt?,
+
+    @SerialName("max_fee")
+    val maxFee: Felt?,
+
     val version: Felt?
 )
 
+data class ContractDefinition(
+    val contract: String
+) {
+    fun parseContract(): Triple<JsonElement, JsonElement, JsonElement> {
+        val compiledContract = Json.parseToJsonElement(this.contract).jsonObject
+        val program = compiledContract["program"]!!
+        val entryPointsByType = compiledContract["entry_points_by_type"]!!
+        val abi = compiledContract["abi"] ?: JsonArray(emptyList())
```

Comment:
We have to ensure those keys are present in the json.

Reply:
I can add some validation if you think that's better. However with the current RPC and Gateway responses, program and entryPointsByType are always present.

Reply:
Please add it

Reply:
Users will get an error that is more meaningful than NullPointerException - everybody hates that

Reply:
Added basic validation and a custom exception for that.